### PR TITLE
[WIP]create init framework for feature store

### PR DESCRIPTION
**Pull request by STALLAAAA**

### CODE REVIEW

File: featurestore/feature_extractor/base_extractor.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Warning: Empty shell for code to be implemented in other class.
+        """
+    @abstractmethod
+    def extract_features(self):
+        """extract and save features.
+
+        Warning: Empty shell for code to be implemented in other class.
+        """
+    @abstractmethod
+    def _save_features(self):
```
Comment (huangyz0918) [inline featurestore/feature_extractor/base_extractor.py:23]: If you want to expose the interface to developers, why name the functions as a build-in scheme?

File: featurestore/feature_extractor/tabular_feature/tabnet.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from featurestore.feature_extractor.base_extractor import BaseFeatureExtractor
+from omegaconf import DictConfig
+
+
+class TabnetFeatureExtractor(BaseFeatureExtractor):
```
Comment (huangyz0918) [inline featurestore/feature_extractor/tabular_feature/tabnet.py:5]: I'm not sure `<Mode Name>FeatureExtractor` is a flexible way for developers and researchers to use. For example, for TabNet, the feature selection is a part of tabular data encoder, and is tight coupling with the model structure. From my point of view, a general tabular data feature processor is more suitable here. The same as other classes.

File: featurestore/feature_extractor/base_extractor.py (hunk 1 above)
Comment (STALLAAAA) [reply featurestore/feature_extractor/base_extractor.py:23]: My original idea was that by using protected abstract functions, developers would be made to integrate saving related action  
or data process actions, which would be good for coding style. Do you have a better idea to accomplish this thought? Or shall we just let it be?

**Comment by HuaizhengZhang** [comment]
A nice PR.

I suggest also drafting a design figure at the same time, so we can discuss it.
image

File: featurestore/feature_extractor/base_extractor.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
```
Comment (HuaizhengZhang) [inline featurestore/feature_extractor/base_extractor.py:1]: If users want to use a model that newly is registered into our model hub. How to easily utilize it for feature extraction

File: featurestore/feature_extractor/tabular_feature/tabnet.py (hunk 2 above)
Comment (STALLAAAA) [reply featurestore/feature_extractor/tabular_feature/tabnet.py:5]: Understand. I am reconsidering this design and may try a better idea after I finish some demos and having a better understanding of details of feature extracting.

File: featurestore/feature_extractor/base_extractor.py (hunk 1 above)
Comment (YuanmingLeee) [reply featurestore/feature_extractor/base_extractor.py:23]: If you use `abstractmethod`, you mean the developer has to change (override) such function in their extension. So, the function should not be protected.

**Comment by HuaizhengZhang** [comment]
Please rename the branch name with a more clear meaning.

File: featurestore/feature_extractor/base_extractor.py (hunk 3 above)
Comment (STALLAAAA) [reply featurestore/feature_extractor/base_extractor.py:1]: From the very first I was counting on users to rewrite the `BaseFeatureExtractor` themselves with some utils provided by us, which means we can barely help users with custom extractor design. Now I am reconsidering this, and after I refine more common parts in feature extraction, I will put up a more customizable solution.

**Pull request closed without merging**